ffi/derive/src/wrapper.rs

```diff
@@ -29,36 +117,27 @@ fn gen_shared_fns(name: &Ident, attrs: &[syn::Attribute]) -> Vec<TokenStream> {
         }
 
         if let syn::Meta::List(derives) = attr.parse_meta().expect("Derive macro invalid") {
-            let mut valid_derives = vec![];
-
-            let mut derive_eq = false;
-            let mut derive_ord = false;
             for derive in derives.nested {
                 if let syn::NestedMeta::Meta(meta) = &derive {
                     if let syn::Meta::Path(path) = meta {
-                        if path.is_ident("Clone") {
-                            shared_fn_impls.push(quote! {
-                                impl Clone for #name {
-                                    fn clone(&self) -> Self {
-                                        let mut output = core::mem::MaybeUninit::uninit();
-
-                                        let handle_id = iroha_ffi::FfiConvert::into_ffi(<#name as iroha_ffi::Handle>::ID, &mut ());
-                                        let clone_result = unsafe { crate::__clone(handle_id, self.__opaque_ptr, output.as_mut_ptr()) };
-
-                                        if clone_result != iroha_ffi::FfiReturn::Ok  {
-                                            panic!("Clone returned: {}", clone_result);
-                                        }
-
-                                        unsafe {iroha_ffi::FfiOutPtrRead::try_read_out(output.assume_init()).expect("Invalid output")}
-                                    }
-                                }
-                            });
+                        if path.is_ident("Copy") {
+                        } else if path.is_ident("Clone") {
+                            shared_fn_impls.push(impl_clone_for_opaque(name));
+                        } else if path.is_ident("Default") {
+                            shared_fn_impls.push(impl_default_for_opaque(name));
+                        } else if path.is_ident("PartialEq") {
+                            shared_fn_impls.push(impl_partial_eq_for_opaque(name));
                         } else if path.is_ident("Eq") {
-                            derive_eq = true;
+                            shared_fn_impls.push(impl_eq_for_opaque(name));
+                        } else if path.is_ident("PartialOrd") {
+                            shared_fn_impls.push(impl_partial_ord_for_opaque(name));
                         } else if path.is_ident("Ord") {
-                            derive_ord = true;
-                        } else if path.is_ident("PartialEq") || path.is_ident("PartialOrd") {
-                            // NOTE: These should be skipped
+                            shared_fn_impls.push(impl_ord_for_opaque(name));
+                        } else if path.is_ident("Setters")
+                            || path.is_ident("Getters")
+                            || path.is_ident("MutGetters")
+                        {
+                            // TODO: What to do about getters/setters?
```

Comment:
What's wrong with providing `getters/setters` for opaque type?

Reply:
I'm still working on this. They will be provided